### PR TITLE
optimizing-memn2n - TF1 -> TF2 migration

### PR DESCRIPTION
**Pull request by mchockal3**
TF1->TF2 migration for optimizing-memn2n repo

### CODE REVIEW

File: optimizing-memn2n/virtual_env_setup/python3_memn2n_tf_env_cpu_requirements.txt [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
+idna==3.4
+keras==2.10.0
+numpy==1.23.5
+protobuf==3.19.6
+python-dateutil==2.8.2
+requests==2.28.1
+scikit-learn==1.2.0
+scipy==1.9.3
+sigopt==8.6.3
+six==1.16.0
```
Comment (harvey-intel) [inline optimizing-memn2n/virtual_env_setup/python3_memn2n_tf_env_cpu_requirements.txt:17]: `six` is for py2<->py3 compatibility, so this is probably not needed anymore?

File: optimizing-memn2n/virtual_env_setup/python3_memn2n_tf_env_cpu_requirements.txt [hunk 2]
```diff
@@ -1,31 +1,23 @@
...
+sklearn==0.0.post1
+tensorboard==2.10.1
+tensorflow-estimator==2.10.0
+tensorflow-macos==2.10.0
```
Comment (harvey-intel) [inline optimizing-memn2n/virtual_env_setup/python3_memn2n_tf_env_cpu_requirements.txt:21]: Can we change this back to just `tensorflow`?

File: optimizing-memn2n/virtual_env_setup/python3_memn2n_tf_env_cpu_requirements.txt [hunk 3]
```diff
@@ -1,31 +1,23 @@
...
-sigopt==3.6.0
-six==1.10.0
-tensorboard==1.13.1
-tensorflow>=1.15.4
-tensorflow-estimator==1.13.0
+gast==0.4.0
+grpcio==1.51.1
+h5py==3.7.0
+idna==3.4
+keras==2.10.0
```
Comment (harvey-intel) [inline optimizing-memn2n/virtual_env_setup/python3_memn2n_tf_env_cpu_requirements.txt:9]: I believe `keras` is part of TF2 now, so don't need separate req here?

I'm guessing `Keras-Preprocessing` is also included as part of TF2

File: optimizing-memn2n/virtual_env_setup/python3_memn2n_tf_env_cpu_requirements.txt [hunk 4]
```diff
@@ -1,31 +1,23 @@
...
+numpy==1.23.5
+protobuf==3.19.6
+python-dateutil==2.8.2
+requests==2.28.1
+scikit-learn==1.2.0
+scipy==1.9.3
+sigopt==8.6.3
+six==1.16.0
+sklearn==0.0.post1
```
Comment (harvey-intel) [inline optimizing-memn2n/virtual_env_setup/python3_memn2n_tf_env_cpu_requirements.txt:18]: instead of sklearn, this package should be `scikit-learn`

**Review by harvey-intel — COMMENTED** [review]
Mostly LGTM, just some minor thoughts on further simplifying the requirements.

**Comment by mchockal3** [comment]
@harvey-intel - I cleaned up the requirements file based on your comments. Also made corrections to README.md that @tskelton-intel pointed to.

**Comment by harvey-intel** [comment]
Can you revert commit `fc977g3` before merging? And open up a separate branch/PR for that change.

**Comment by mchockal3** [comment]
> Can you revert commit `fc977g3` before merging? And open up a separate branch/PR for that change.

Sorry about that - reverted all the other changes now.

**Review by harvey-intel — APPROVED** [review]
Thanks!

**Pull request merged**